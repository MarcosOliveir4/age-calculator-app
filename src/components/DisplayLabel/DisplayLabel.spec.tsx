import { render, screen } from '@testing-library/react';
import { DisplayLabel, DisplayLabelInterface } from './index';

const defaultProps: DisplayLabelInterface = {
  label: 'years',
  value: 22,
};

const sut = (props = defaultProps) => {
  return render(<DisplayLabel {...props} />);
};

describe('<DisplayLabel />', () => {
  it('should render the component', () => {
    sut();
    const screenValue = screen.getByTestId('display-value');
    const screenLabel = screen.getByTestId('display-label');
    expect(screenLabel).toHaveTextContent(defaultProps.label);
    expect(screenValue).toHaveTextContent(`${defaultProps.value}`);
  });
  it('should render component with validation of no value', () => {
    sut({ label: 'days' });
    const screenValue = screen.getByTestId('display-value');
    expect(screenValue).toHaveTextContent('- -');
  });
  it('should render snapshot', () => {
    sut();
    expect(screen.getByTestId('display')).toMatchSnapshot();
  });
});
