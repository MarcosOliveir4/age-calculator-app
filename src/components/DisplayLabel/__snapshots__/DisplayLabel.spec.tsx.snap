// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayLabel /> should render snapshot 1`] = `
<div
  class="sc-beqWaB gXTDLZ"
  data-testid="display"
>
  <span
    class="sc-gueYoa eaFbDW"
    color="--purple"
    data-testid="display-value"
  >
    22
  </span>
  <span
    class="sc-gueYoa hReXcv"
    color="--black"
    data-testid="display-label"
  >
    years
  </span>
</div>
`;
